src/app/app-routing.module.ts

```diff
@@ -1,10 +1,24 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
-const routes: Routes = [];
+import { AppComponent } from './app.component';
+import { HeaderComponent } from './components/header/header.component';
+import { LoginComponent } from './components/header/login/login.component';
+import { HomepageComponent } from './components/homepage/homepage.component';
+import { ListingpageComponent } from './components/listingpage/listingpage.component';
+import { ProductpageComponent } from './components/productpage/productpage.component';
+
+
+const routes: Routes = [
+  {path : "", component : HomepageComponent},
+  {path : "Home", component : HomepageComponent},
```

Comment:
there should be a single path for the HomepageComponent

Reply:
Deleted line 14